packages/medusa/src/api/routes/admin/product-categories/list-product-categories.ts

```diff
@@ -0,0 +1,118 @@
+import { Request, Response } from "express"
+
+import { ProductCategoryService } from "../../../../services"
+import { Type } from "class-transformer"
+
+/**
+ * @oas [get] /product-categories
+ * operationId: "GetProductCategories"
+ * summary: "List Product Categories"
+ * description: "Retrieve a list of product categories."
+ * x-authenticated: true
+ * parameters:
+ *   - (query) q {string} Query used for searching product category names orhandles.
+ *   - (query) offset=0 {integer} How many groups to skip in the result.
+ *   - (query) limit=10 {integer} Limit the number of product categories returned.
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.productCategories.list()
+ *       .then(({ product_category, limit, offset, count }) => {
+ *         console.log(product_category.length);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request GET 'https://medusa-url.com/admin/product-categories' \
+ *       --header 'Authorization: Bearer {api_token}'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Product Categories
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           type: object
+ *           properties:
+ *             product_category:
+ *               type: array
+ *               items:
+ *                 $ref: "#/components/schemas/ProductCategory"
+ *             count:
+ *               type: integer
+ *               description: The total number of items available
+ *             offset:
+ *               type: integer
+ *               description: The number of items skipped before these items
+ *             limit:
+ *               type: integer
+ *               description: The number of items per page
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req: Request, res: Response) => {
+  const productCategoryService: ProductCategoryService = req.scope.resolve(
+    "productCategoryService"
+  )
+
+  const [data, count] = await productCategoryService.listAndCount(
+    req.filterableFields,
+    req.listConfig
+  )
+
+  const { limit, offset } = req.validatedQuery
+
+  res.json({
+    count,
+    product_categories: data,
+    offset,
+    limit,
+  })
+}
+
+export class AdminGetProductCategoriesParams {
```

Comment:
todo: extends `FindPaginationParams` and remove offset and limit. If it make sense, you can look at `extendedFindParamsMixin`